### PR TITLE
Nested spread directive

### LINKED ISSUE #857: Nested @spread directive not working
**Describe the bug**

When I use the `@spread` directive at the top level (`createOrder(input: OrderInput @spread): Order @create` I am getting the expected behaviour. However, if I use a nested `@spread` directive inside `OrderInput`, only the one at the top level is being spread out.

**Expected behavior/Solution**

Everytime there is the `@spread` directive, its arguments should be flattened. E.g.
```
input ProductNestedInput {
    name: String!
}

input ProductInput {
    product: ProductNestedInput! @spread
}

type Mutation {
    createProduct(input: ProductInput @spread): Product @create
}
```
The previous mutation should allow me to create an instance of the following type:
```
type Product {
    id: ID!
    name: String!
}
```
I have used Lighthouse for very little time, so I can't tell the reason for this behavior, but I think it has something to do with either `@spread` only being parsed when in the top level or with `@spread` being parsed after it has attempted to create the instance of an object.

**Steps to reproduce**

1. Create a Product model with only the name field:
```
use Illuminate\Database\Eloquent\Model;

class Product extends Model
{
    protected $fillable = ['name'];
}
```
2. Write the following in the schema file:
```
type Product {
    id: ID!
    name: String!
}

input ProductNestedInput {
    name: String!
}

input ProductInput {
    product: ProductNestedInput! @spread
}

type Mutation {
    createProduct(input: ProductInput @spread): Product @create
}
```
3. Call the following GraphQL Query:
```
mutation {
  createProduct(input:{
    product: {
      name: "Test"
    }
  }) {
    id
    name
  }
}
```

**Output/Logs**

<details><summary>Click to expand</summary>

```
{
  "errors": [
    {
      "debugMessage": "SQLSTATE[HY000]: General error: 1364 Field 'name' doesn't have a default value (SQL: insert into `products` (`updated_at`, `created_at`) values (2019-07-10 18:36:29, 2019-07-10 18:36:29))",
      "message": "Internal server error",
      "extensions": {
        "category": "internal"
      },
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "createProduct"
      ],
      "trace": [
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Connection.php",
          "line": 624,
          "call": "Illuminate\\Database\\Connection::runQueryCallback('insert into `products` (`updated_at`, `created_at`) values (?, ?)', array(2), instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Connection.php",
          "line": 459,
          "call": "Illuminate\\Database\\Connection::run('insert into `products` (`updated_at`, `created_at`) values (?, ?)', array(2), instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Connection.php",
          "line": 411,
          "call": "Illuminate\\Database\\Connection::statement('insert into `products` (`updated_at`, `created_at`) values (?, ?)', array(2))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Query\/Processors\/Processor.php",
          "line": 32,
          "call": "Illuminate\\Database\\Connection::insert('insert into `products` (`updated_at`, `created_at`) values (?, ?)', array(2))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Query\/Builder.php",
          "line": 2657,
          "call": "Illuminate\\Database\\Query\\Processors\\Processor::processInsertGetId(instance of Illuminate\\Database\\Query\\Builder, 'insert into `products` (`updated_at`, `created_at`) values (?, ?)', array(2), 'id')"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Eloquent\/Builder.php",
          "line": 1353,
          "call": "Illuminate\\Database\\Query\\Builder::insertGetId(array(2), 'id')"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Eloquent\/Model.php",
          "line": 839,
          "call": "Illuminate\\Database\\Eloquent\\Builder::__call('insertGetId', array(2))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Eloquent\/Model.php",
          "line": 804,
          "call": "Illuminate\\Database\\Eloquent\\Model::insertAndSetId(instance of Illuminate\\Database\\Eloquent\\Builder, array(2))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Eloquent\/Model.php",
          "line": 667,
          "call": "Illuminate\\Database\\Eloquent\\Model::performInsert(instance of Illuminate\\Database\\Eloquent\\Builder)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/nuwave\/lighthouse\/src\/Execution\/MutationExecutor.php",
          "line": 173,
          "call": "Illuminate\\Database\\Eloquent\\Model::save()"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/nuwave\/lighthouse\/src\/Execution\/MutationExecutor.php",
          "line": 47,
          "call": "Nuwave\\Lighthouse\\Execution\\MutationExecutor::saveModelWithPotentialParent(instance of App\\Product, instance of Illuminate\\Support\\Collection(1), null)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/nuwave\/lighthouse\/src\/Schema\/Directives\/CreateDirective.php",
          "line": 60,
          "call": "Nuwave\\Lighthouse\\Execution\\MutationExecutor::executeCreate(instance of App\\Product, instance of Illuminate\\Support\\Collection(1))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Concerns\/ManagesTransactions.php",
          "line": 29,
          "call": "Nuwave\\Lighthouse\\Schema\\Directives\\CreateDirective::Nuwave\\Lighthouse\\Schema\\Directives\\{closure}(instance of Illuminate\\Database\\MySqlConnection)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/nuwave\/lighthouse\/src\/Schema\/Directives\/CreateDirective.php",
          "line": 64,
          "call": "Illuminate\\Database\\Connection::transaction(instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/nuwave\/lighthouse\/src\/Schema\/Factories\/FieldFactory.php",
          "line": 221,
          "call": "Nuwave\\Lighthouse\\Schema\\Directives\\CreateDirective::Nuwave\\Lighthouse\\Schema\\Directives\\{closure}(null, array(1), instance of Nuwave\\Lighthouse\\Schema\\Context, instance of GraphQL\\Type\\Definition\\ResolveInfo)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/Executor\/ReferenceExecutor.php",
          "line": 632,
          "call": "Nuwave\\Lighthouse\\Schema\\Factories\\FieldFactory::Nuwave\\Lighthouse\\Schema\\Factories\\{closure}(null, array(1), instance of Nuwave\\Lighthouse\\Schema\\Context, instance of GraphQL\\Type\\Definition\\ResolveInfo)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/Executor\/ReferenceExecutor.php",
          "line": 555,
          "call": "GraphQL\\Executor\\ReferenceExecutor::resolveOrError(instance of GraphQL\\Type\\Definition\\FieldDefinition, instance of GraphQL\\Language\\AST\\FieldNode, instance of Closure, null, instance of Nuwave\\Lighthouse\\Schema\\Context, instance of GraphQL\\Type\\Definition\\ResolveInfo)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/Executor\/ReferenceExecutor.php",
          "line": 478,
          "call": "GraphQL\\Executor\\ReferenceExecutor::resolveField(GraphQLType: Mutation, null, instance of ArrayObject(1), array(1))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/Executor\/ReferenceExecutor.php",
          "line": 921,
          "call": "GraphQL\\Executor\\ReferenceExecutor::GraphQL\\Executor\\{closure}(array(0), 'createProduct')"
        },
        {
          "call": "GraphQL\\Executor\\ReferenceExecutor::GraphQL\\Executor\\{closure}(array(0), 'createProduct')"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/Executor\/ReferenceExecutor.php",
          "line": 923,
          "function": "array_reduce(array(1), instance of Closure, array(0))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/Executor\/ReferenceExecutor.php",
          "line": 494,
          "call": "GraphQL\\Executor\\ReferenceExecutor::promiseReduce(array(1), instance of Closure, array(0))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/Executor\/ReferenceExecutor.php",
          "line": 256,
          "call": "GraphQL\\Executor\\ReferenceExecutor::executeFieldsSerially(GraphQLType: Mutation, null, array(0), instance of ArrayObject(1))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/Executor\/ReferenceExecutor.php",
          "line": 208,
          "call": "GraphQL\\Executor\\ReferenceExecutor::executeOperation(instance of GraphQL\\Language\\AST\\OperationDefinitionNode, null)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/Executor\/Executor.php",
          "line": 155,
          "call": "GraphQL\\Executor\\ReferenceExecutor::doExecute()"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/GraphQL.php",
          "line": 165,
          "call": "GraphQL\\Executor\\Executor::promiseToExecute(instance of GraphQL\\Executor\\Promise\\Adapter\\SyncPromiseAdapter, instance of GraphQL\\Type\\Schema, instance of GraphQL\\Language\\AST\\DocumentNode, null, instance of Nuwave\\Lighthouse\\Schema\\Context, array(0), null, null)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/webonyx\/graphql-php\/src\/GraphQL.php",
          "line": 90,
          "call": "GraphQL\\GraphQL::promiseToExecute(instance of GraphQL\\Executor\\Promise\\Adapter\\SyncPromiseAdapter, instance of GraphQL\\Type\\Schema, 'mutation {\n  createProduct(input:{\n    product: {\n      name: \"Test\"\n    }\n  }) {\n    id\n    name\n  }\n}', null, instance of Nuwave\\Lighthouse\\Schema\\Context, array(0), null, null, array(29))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/nuwave\/lighthouse\/src\/GraphQL.php",
          "line": 191,
          "call": "GraphQL\\GraphQL::executeQuery(instance of GraphQL\\Type\\Schema, 'mutation {\n  createProduct(input:{\n    product: {\n      name: \"Test\"\n    }\n  }) {\n    id\n    name\n  }\n}', null, instance of Nuwave\\Lighthouse\\Schema\\Context, array(0), null, null, array(29))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/nuwave\/lighthouse\/src\/GraphQL.php",
          "line": 130,
          "call": "Nuwave\\Lighthouse\\GraphQL::executeQuery('mutation {\n  createProduct(input:{\n    product: {\n      name: \"Test\"\n    }\n  }) {\n    id\n    name\n  }\n}', instance of Nuwave\\Lighthouse\\Schema\\Context, array(0), null, null)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/nuwave\/lighthouse\/src\/Support\/Http\/Controllers\/GraphQLController.php",
          "line": 79,
          "call": "Nuwave\\Lighthouse\\GraphQL::executeRequest(instance of Nuwave\\Lighthouse\\Execution\\LighthouseRequest)"
        },
        {
          "call": "Nuwave\\Lighthouse\\Support\\Http\\Controllers\\GraphQLController::query(instance of Nuwave\\Lighthouse\\Execution\\LighthouseRequest)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Controller.php",
          "line": 54,
          "function": "call_user_func_array(array(2), array(1))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/ControllerDispatcher.php",
          "line": 45,
          "call": "Illuminate\\Routing\\Controller::callAction('query', array(1))"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Route.php",
          "line": 219,
          "call": "Illuminate\\Routing\\ControllerDispatcher::dispatch(instance of Illuminate\\Routing\\Route, instance of Nuwave\\Lighthouse\\Support\\Http\\Controllers\\GraphQLController, 'query')"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Route.php",
          "line": 176,
          "call": "Illuminate\\Routing\\Route::runController()"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Router.php",
          "line": 680,
          "call": "Illuminate\\Routing\\Route::run()"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Pipeline.php",
          "line": 30,
          "call": "Illuminate\\Routing\\Router::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/nuwave\/lighthouse\/src\/Support\/Http\/Middleware\/AcceptJson.php",
          "line": 30,
          "call": "Illuminate\\Routing\\Pipeline::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php",
          "line": 163,
          "call": "Nuwave\\Lighthouse\\Support\\Http\\Middleware\\AcceptJson::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Pipeline.php",
          "line": 53,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php",
          "line": 104,
          "call": "Illuminate\\Routing\\Pipeline::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Router.php",
          "line": 682,
          "call": "Illuminate\\Pipeline\\Pipeline::then(instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Router.php",
          "line": 657,
          "call": "Illuminate\\Routing\\Router::runRouteWithinStack(instance of Illuminate\\Routing\\Route, instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Router.php",
          "line": 623,
          "call": "Illuminate\\Routing\\Router::runRoute(instance of Illuminate\\Http\\Request, instance of Illuminate\\Routing\\Route)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Router.php",
          "line": 612,
          "call": "Illuminate\\Routing\\Router::dispatchToRoute(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Http\/Kernel.php",
          "line": 176,
          "call": "Illuminate\\Routing\\Router::dispatch(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Pipeline.php",
          "line": 30,
          "call": "Illuminate\\Foundation\\Http\\Kernel::Illuminate\\Foundation\\Http\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/fideloper\/proxy\/src\/TrustProxies.php",
          "line": 57,
          "call": "Illuminate\\Routing\\Pipeline::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php",
          "line": 163,
          "call": "Fideloper\\Proxy\\TrustProxies::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Pipeline.php",
          "line": 53,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Http\/Middleware\/TransformsRequest.php",
          "line": 21,
          "call": "Illuminate\\Routing\\Pipeline::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php",
          "line": 163,
          "call": "Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Pipeline.php",
          "line": 53,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Http\/Middleware\/TransformsRequest.php",
          "line": 21,
          "call": "Illuminate\\Routing\\Pipeline::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php",
          "line": 163,
          "call": "Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Pipeline.php",
          "line": 53,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Http\/Middleware\/ValidatePostSize.php",
          "line": 27,
          "call": "Illuminate\\Routing\\Pipeline::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php",
          "line": 163,
          "call": "Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Pipeline.php",
          "line": 53,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Http\/Middleware\/CheckForMaintenanceMode.php",
          "line": 62,
          "call": "Illuminate\\Routing\\Pipeline::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php",
          "line": 163,
          "call": "Illuminate\\Foundation\\Http\\Middleware\\CheckForMaintenanceMode::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Pipeline.php",
          "line": 53,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php",
          "line": 104,
          "call": "Illuminate\\Routing\\Pipeline::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Http\/Kernel.php",
          "line": 151,
          "call": "Illuminate\\Pipeline\\Pipeline::then(instance of Closure)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Http\/Kernel.php",
          "line": 116,
          "call": "Illuminate\\Foundation\\Http\\Kernel::sendRequestThroughRouter(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/public\/index.php",
          "line": 55,
          "call": "Illuminate\\Foundation\\Http\\Kernel::handle(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "\/home\/jpyamamoto\/Code\/php\/test\/server.php",
          "line": 21,
          "function": "require_once('\/home\/jpyamamoto\/Code\/php\/test\/public\/index.php')"
        }
      ]
    }
  ],
  "data": {
    "createProduct": null
  }
}
```

</details></br>

**Note**
I know that for this specific example it is stupid to call a nested input like that. However, I just came up with this simple test case in order to demonstrate a similar scenario that I'm facing where it actually makes sense to use a nested input that requires the `@spread` directive.

**Environment**

Lighthouse Version: 3.7.0
Laravel Version: 5.8.27

### PR DESCRIPTION
**Pull request by spawnia**
- [x] Added or updated tests
- [x] Added Docs for all relevant versions
- [x] Updated the changelog

Resolves #857

**Changes**

This allows to use the `@spread` directive nested within a given input.

```graphql
type Mutation {
    updatePost(
        id: ID!
        input: PostInput! @spread
    ): Post @update
}

input PostInput {
    title: String!
    content: PostContent @spread
}

input PostContent {
    imageUrl: String
}
```

is now flattened to

```php
[
    'id' => 12,
    'title' => 'My awesome title',
    'imageUrl' = 'http://some.site/image.jpg',
]
```

**Breaking changes**

Some internals were shifted around, but there are no user-facing changes.

### CODE REVIEW

**Comment by spawnia** [comment]
@JPYamamoto can you take a look?

File: src/Execution/Arguments/Argument.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        return $this->toPlainRecursive($this->value);
+    }
+
+    /**
+     * Convert the given value to plain PHP values recursively.
+     *
+     * @param  \Nuwave\Lighthouse\Execution\Arguments\ArgumentSet|\Nuwave\Lighthouse\Execution\Arguments\ArgumentSet[]|mixed|mixed[]  $value
+     * @return mixed|mixed[]
+     */
+    protected function toPlainRecursive($value)
```
Comment (olivernybroe) [inline src/Execution/Arguments/Argument.php]: This could be static 😃

File: src/Execution/Arguments/TypedArgs.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    /**
+     * Wrap client-given args with type information.
+     *
+     * @param  mixed[]  $args
+     * @param  \GraphQL\Language\AST\NodeList|\GraphQL\Language\AST\InputValueDefinitionNode[] $argumentDefinitions
+     * @return \Nuwave\Lighthouse\Execution\Arguments\Argument[]
+     */
+    protected function wrapArgs(array $args, $argumentDefinitions): array
```
Comment (olivernybroe) [inline src/Execution/Arguments/TypedArgs.php]: I think this method could use some love. It's kinda messy and big.
Just if there are any obvious refactoring you can do here :)

File: src/Schema/ExecutableTypeNodeConverter.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                return Type::int();
+            case 'Boolean':
+                return Type::boolean();
+            case 'Float':
+                return Type::float();
+            case 'String':
+                return Type::string();
+            default:
+                return $this->typeRegistry->get($nodeName);
+        }
```
Comment (olivernybroe) [inline src/Schema/ExecutableTypeNodeConverter.php]: Replace with `Type::getStandardTypes()[$nodeName] ?? $this->typeRegistry->get($nodeName);`

Comment (olivernybroe) [reply src/Schema/ExecutableTypeNodeConverter.php]: It wouldn't allow me to make multi line suggestions

Comment (spawnia) [reply src/Schema/ExecutableTypeNodeConverter.php]: nice!

File: src/Execution/Arguments/TypedArgs.php (hunk 2 above)
Comment (spawnia) [reply src/Execution/Arguments/TypedArgs.php]: Good idea, i broke it down.

Comment (olivernybroe) [reply src/Execution/Arguments/TypedArgs.php]: Yeah, much nicer 👍 Awesome work

**Comment by spawnia** [comment]
@olivernybroe there is a bigger picture behind this change: I wanted to have a general purpose data structure that contains both arguments and type information, reduced down to what actually matters when processing args.

I plan to use this to unravel more of the complexity in the `FieldFactory` and ultimately work towards https://github.com/nuwave/lighthouse/pull/899.

**Comment by spawnia** [comment]
> I don't really have the time to test it out and see if it functions properly. So in that part I'll just trust you or hope someone else will do that part of the review.

I am relying on the test suite, it covers all the changes quite thorough.

**Pull request merged**